lib/commands/hasFocus.js

```diff
@@ -26,23 +26,34 @@
  *
  */
 
+import { RuntimeError } from '../utils/ErrorHandler'
+
 let hasFocus = function (selector) {
-    const finalSelector = selector === null ? this.lastPromise.inspect().value.selector : selector
-    let result = this.execute(function (selector) {
-        var focused = document.activeElement
-        if (!focused) {
-            return false
+    let result = this.elements(selector).then((res) => {
+        /**
+         * check if element was found and throw error if not
+         */
+        if (!res.value) {
+            throw new RuntimeError(7)
         }
 
-        var instanceOfSelectorHasFocus = false
-        var selectorArray = document.querySelectorAll(selector)
-        for (var i = 0; i < selectorArray.length; i++) {
-            if (focused === selectorArray[i]) {
-                instanceOfSelectorHasFocus = true
+        return res.value
+    }).then((elements) => {
+        return this.execute(function (selectorArray) {
+            var focused = document.activeElement
+            if (!focused) {
+                return false
             }
-        }
-        return instanceOfSelectorHasFocus
-    }, finalSelector)
+
+            var instanceOfSelectorHasFocus = false
+            for (var i = 0; i < selectorArray.length; i++) {
```

Comment:
Also let's rename `selectorArray` to `elemArray` or something along these lines.